dhis2/dhis2-core
# feat: tracker import errors to job tracking error forward [DHIS2-15276]

### Summary
Adjustments to the job progress error tracking for tracker import jobs. 
This does add error transfer from the `ValidationReport` to the job progress errors.

Since two different enums are used as source of error codes, `ErrorCode` and `ValidationCode` the job progress API got `addError` methods which accept one or the other. Internally the code is then stored as `String`.
A client can distinguish the codes based on the added `codes` property to the root object of `/api/jobConfigurations/errors`.
The value can be understood as a sort of namespace or part of a translation key root or path segment.

The input that caused an error can now be included using `includeInput` flag parameter, for example: `/api/jobConfigurations/errors?includeInput=true`

The tracker input file storage format changed from binary (java serialisation) to JSON so it can be included in the JSON as described above.

The tracker `Error` created from a validation error now included the plain argument list as this is needed when transferring errors to the job progress error tracking which does not store the message but adds it on the fly for API requests based on the error code and the args.

### Automatic Testing
Existing tests got adjusted, mostly to provide the args list. 

A new test scenario was added that runs a tracker import with error and checks the error endpoint with the `includeInput` to verify both that tracker errors are transferred and that the input inclusion works.

### Manual Testing 
* create a tracker import: POST to `/api/tracker?async=true` with a JSON payload with errors, for example 
```json
{
    "trackedEntities": [
        {
            "trackedEntity":"sHH8mh1Fn0z",
            "trackedEntityType": "nEenWmSyUEp",
            "orgUnit": "DiszpKrYNg8"
        }
    ]
}
```
* find the created job, e.g. `api/jobConfigurations/gist?filter=jobType:eq:TRACKER_IMPORT_JOB&order=created:desc`
* check the errors `api/jobConfigurations/errors?includeInput=true` has an entry for the job and also includes the input JSON as `input` (note that this is **not** 1:1 of the original JSON send but a JSON for a intermediate format based on `TrackerObjects` type)
